programs/bpf_loader/src/syscalls/mod.rs

```diff
@@ -1880,6 +1771,71 @@ declare_syscall!(
     }
 );
 
+declare_syscallhash!(
+    // Generic Hashing Syscall
+    SyscallHash,
+    fn inner_call<H: HasherImpl>(
+        invoke_context: &mut InvokeContext,
+        vals_addr: u64,
+        vals_len: u64,
+        result_addr: u64,
+        _arg4: u64,
+        _arg5: u64,
+        memory_mapping: &mut MemoryMapping,
+    ) -> Result<u64, Error> {
+        let compute_budget = invoke_context.get_compute_budget();
+        if compute_budget.sha256_max_slices < vals_len {
```

Comment:
we can't just use the sha256 compute budget values for all digests.  they should be queried with another method on the trait.

Reply:
Done!

Reply:
so I went with the default values of sha256 in compute_budget variables for the other hashes as well since I didn't know which values they are supposed to use as previously they were using the same ones as sha256 so if they need to be updated lmk.